brave/README.md

```diff
@@ -167,37 +167,59 @@ void userCode() {
 }
 ```
 
-### RPC tracing
+### Request tracing
 Check for [instrumentation written here](../instrumentation/) and [Zipkin's list](https://zipkin.io/pages/existing_instrumentations.html)
-before rolling your own RPC instrumentation!
-
-RPC tracing is often done automatically by interceptors. Under the scenes,
-they add tags and events that relate to their role in an RPC operation.
-
-Note: this is intentionally not an HTTP example, as we have [a special layer](../instrumentation/http)
-for that.
-
-Here's an example of a client span:
-```java
-// before you send a request, add metadata that describes the operation
-span = tracer.nextSpan().name(service + "/" + method).kind(CLIENT);
-span.tag("myrpc.version", "1.0.0");
-span.remoteServiceName("backend");
-span.remoteIpAndPort("172.3.4.1", 8108);
-
-// Add the trace context to the request, so it can be propagated in-band
-tracing.propagation().injector(Request::addHeader)
-                     .inject(span.context(), request);
-
-// when the request is scheduled, start the span
+before rolling your own Request instrumentation!
+
+Brave includes two fundamental types to support tracing interprocess
+communication `brave.Request` and `brave.Response`. These types represent the
+following communication patterns, defined by the [Zipkin Api][https://zipkin.io/zipkin-api/#/default/post_spans]
+and returned by `spanKind()`:
+
+ * CLIENT
+ * SERVER
+ * PRODUCER
+ * CONSUMER
+
+Once you model your request, you generally need to...
+1. Start the span and add trace headers to the request
+2. Put the span in scope so things like log integration works
+3. Invoke the request
+4. Catch any errors
+5. Complete the span
+
+Here's a simplified example:
+```java
+requestWrapper = new ClientRequestWrapper(request);
+span = tracer.nextSpan(sampler, requestWrapper); // 1.
+tracing.propagation().injector(ClientRequestWrapper::addHeader)
+                     .inject(span.context(), requestWrapper);
+span.kind(request.spanKind());
+span.name("Report");
 span.start();
+try (Scope ws = currentTraceContext.newScope(span.context())) { // 2.
+  return invoke(request); // 3.
+} catch (Throwable e) {
+  span.error(error); // 4.
+  throw e;
+} finally {
+  span.finish(); // 5.
+}
+```
 
-// if there is an error, tag the span
-span.tag("error", error.getCode());
+The above code is example only. Abstractions exist that handle aspects
+including sampling, header processing and data parsing:
+ * [HTTP](../instrumentation/http/README.md) is CLIENT/SERVER
+ * [Messaging](../instrumentation/rpc/README.md) is PRODUCER/CONSUMER
```

Comment:
Wrong href

Reply:
thx fixed in latest.  I realized we repeated a lot of this in the instrumentation README!